File: Unit2RecyclerView/app/src/main/java/com/uit/unit2_recyclerview/MainActivity.java
package com.uit.unit2_recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.view.View;
import android.widget.RadioGroup;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputEditText;
import com.uit.unit2_recyclerview.gridview_spinner.GridViewActivity;

public class MainActivity extends AppCompatActivity implements MainItemClicked{
    RecyclerView mRecyclerView;
    RecyclerViewAdapter mAdapter;
    TextInputEditText mInputName, mInputMaNV;
    MaterialButton mInsertButton;
    AppCompatTextView mPosition, mValue;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        gotoGridActivity();

        setupRecyclerView();
        setupView();
    }

    private void gotoGridActivity(){
        Intent intent = new Intent(this, GridViewActivity.class);
        startActivity(intent);
        finish();
    }

    private void setupView() {
        mInputMaNV = findViewById(R.id.et_input_manv);
        mInputName = findViewById(R.id.et_input_name);
        mInsertButton = findViewById(R.id.btn_insert);
        mPosition = findViewById(R.id.tv_position);
        mValue = findViewById(R.id.tv_value);
    }

    public void buttonInsertClicked(View view){
        Editable name = mInputName.getText();
        if (name != null && !TextUtils.isEmpty(name.toString().trim())){
            addNewEmployee();
            mInputName.setText("");
        }
    }

    private void setupRecyclerView() {
        mRecyclerView = findViewById(R.id.rv_main);
        mAdapter = new RecyclerViewAdapter();
        mRecyclerView.hasFixedSize();
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        mRecyclerView.setLayoutManager(layoutManager);
        mRecyclerView.setAdapter(mAdapter);
        mAdapter.itemClicked = this;
    }

    @Override
    public void onItemClicked(int position, String value) {
        String pos = getString(R.string.position) + position;
        mPosition.setText(pos);
        String val = getString(R.string.value) + " = " + value;
        mValue.setText(val);
    }

    public void addNewEmployee() {
        Employee employee;
        //Lấy ra đúng id của Radio Button được checked
        RadioGroup group = findViewById(R.id.radio_group);
        int radId = group.getCheckedRadioButtonId();
        if (radId == R.id.radio_full) {
            employee = new EmployeeFullTime();
        }else {
            employee = new EmployeePartTime();
        }

        String id = mInputMaNV.getText().toString();
        String name = mInputName.getText().toString();

        //FullTime hay Partime thì cũng là Employee nên có các hàm này là hiển nhiên
        employee.setId(id);
        employee.setName(name);
        //Đưa employee vào ArrayList
        mAdapter.mListNV.add(employee);
        //Cập nhập giao diện
        mAdapter.notifyItemInserted(mAdapter.mListNV.size());
    }
}

File: Unit1LayoutOverview/app/src/main/java/com/uit/unit1_layoutoverview/MainActivity.kt
package com.uit.unit1_layoutoverview

import android.content.Intent
import android.os.Bundle
import android.widget.LinearLayout
import android.widget.LinearLayout.HORIZONTAL
import android.widget.LinearLayout.LayoutParams
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var llNameContainer: LinearLayout
    private lateinit var llAddressContainer: LinearLayout
    private lateinit var llParentContainer: LinearLayout
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        createNameContainer()
        createAddressContainer()
        createParentContainer()
        setContentView(R.layout.activity_main)
//        setContentView(llParentContainer)
        supportActionBar?.title = "Constraint Layout"
    }

    private fun createNameContainer() {
        llNameContainer = LinearLayout(this)
        llNameContainer.layoutParams = LayoutParams(
            LayoutParams.MATCH_PARENT,
            LayoutParams.WRAP_CONTENT
        )
        llNameContainer.orientation = HORIZONTAL

        val tvName = TextView(this)
        tvName.text = "Name: "
        llNameContainer.addView(tvName)

        val tvNameValue = TextView(this)
        tvNameValue.text = "John Doe"
        llNameContainer.addView(tvNameValue)
    }

    private fun createAddressContainer(){
        llAddressContainer = LinearLayout(this)
        llAddressContainer.layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        llAddressContainer.orientation = HORIZONTAL

        val tvAddress = TextView(this)
        tvAddress.text = "Address:"
        llAddressContainer.addView(tvAddress)

        val tvAddressValue = TextView(this)
        tvAddressValue.text = "911 Hollywood Blvd"
        llAddressContainer.addView(tvAddressValue)
    }

    private fun createParentContainer() {
        llParentContainer = LinearLayout(this)

        llParentContainer.layoutParams = LayoutParams(
            LayoutParams.MATCH_PARENT,
            LayoutParams.MATCH_PARENT
        )
        llParentContainer.orientation = LinearLayout.VERTICAL

        llParentContainer.addView(llNameContainer)
        llParentContainer.addView(llAddressContainer)
    }


}
